website/content/en/docs/reference/configuration/unit-tests.md

```diff
@@ -0,0 +1,359 @@
+[comparisons].
+
+[comparisons]: https://vrl.dev/expressions/#comparisons
+[contains]: https://vrl.dev/functions/#contains
+[exists]: https://vrl.dev/functions/#exists
+[includes]: https://vrl.dev/functions/#includes
+[type]: https://vrl.dev/functions/#type-functions
+[vrl]: https://vrl.dev
+{{< /success >}}
+
+### Only the last expression is evaluated
+
+When writing a VRL condition for your test output, it's important to bear in mind that the condition
+passes if the **last expression** provided evaluates to `true`. If you include multiple Boolean
+expressions, all but the last one are disregarded. This condition would thus evaluate to `true`:
+
+```ruby
+1 == 2
+"booper" == "bopper"
+true
+```
+
+Because of this, we recommend always structuring your conditions as a **single Boolean expression**,
+using `&&` (and) and `||` to chain Boolean expressions together when multiple expressions are in
+play. The condition immediately above would evaluate to `false`, as we'd expect, if rewritten like
+this:
+
+```ruby
+1 == 2 && "booper" == "bopper" && true
+```
+
+### Multiple lines
+
+For the sake of readability, you can also spread a single Boolean expression across multiple lines.
+Both of the following are valid:
+
+```ruby
+# Indented
+1 == 2 &&
+  "booper" == "bopper" &&
+  true
+
+# Not indented
+1 == 2 &&
+"booper" == "bopper" &&
+true
+```
+
+## Running unit tests
+
+You can execute tests within a configuration file using the [`test`][vector_test] subcommand:
+
+```bash
+vector test /etc/vector/vector.toml
+```
+
+You can also specify multiple files:
+
+```bash
+vector test /etc/vector/*.toml
+```
+
+Specifying multiple files can be useful if you want to keep your unit tests in a separate file from
+your pipeline configuration. Vector treats the multiple files as a single configuration.
+
+## Example unit test configuration {#example}
+
+Let's start with an annotated unit testing example:
+
+```toml
+# The transform being tested is a Vector Remap Language (VRL) transform that
+# adds two fields to each incoming log event: a timestamp and a unique ID
+[transforms.add_metadata]
+type = "remap"
+inputs = []
+source = '''
+.timestamp = now()
+.id = uuid_v4()
+'''
+
+# Here we begin configuring our test suite
+[[tests]]
+name = "transaction_logging_test"
+
+# The inputs for the test
+[[tests.inputs]]
+insert_at = "add_metadata" # The transform into which the testing event is inserted
+type = "log"               # The event type (either log or metric)
+
+# The test log event that is passed to the `add_metadata` transform
+[tests.inputs.log_fields]
+message = "successful transaction"
+code = 200
+
+# The expected outputs of the test
+[[tests.outputs]]
+extract_from = "add_metadata" # The transform from which the resulting event is extracted
+
+# The declaration of what we expect
+[[tests.outputs.conditions]]
+type = "vrl"
+source = '''
+is_timestamp(.timestamp)
+is_string(.id)
+.message == "successful transaction"
```

Comment:
I believe this isn't quite right since, as mentioned above, only the last boolean will cause the test to pass or fail.

Given, if we go with my assert style tests mentioned above, this point is moot. It would instead look like:

```
assert!(is_timestamp(.timestamp), ".timestamp should be a timestamp")
assert!(is_string(.id), ".id should be a string")
assert_eq!(.message, "successful transaction")
```